src/main/php/util/Objects.class.php

```diff
@@ -54,8 +58,15 @@ public static function compare($a, $b) {
         if (0 !== $r= self::compare($val, $b[$key])) return $r;
       }
       return 0;
+    } else if ($a === $b) {
+      return 0;
+    } else if (is_object($a)) {
+      return (is_object($b) && get_class($a) === get_class($b))
+        ? self::compare((array)$a, (array)$b)
+        : 1
+      ;
     } else {
-      return $a === $b ? 0 : ($a < $b ? -1 : 1);
+      return $a < $b ? -1 : 1;
     }
```

Comment:
Here, we must live with a small performance hit, and cannot make use of lazy left-to-right evaluation. The minimum we can do is to move the `===` check above and short-circuit it before having to invoke `is_object()`.